src/diagnostics_component/utils/file_info_utils/src/file_info_utils.c

```diff
@@ -168,13 +168,11 @@ bool FileInfoUtils_FillFileInfoWithNewestFilesInDir(FileInfo* logFiles, size_t l
 
     if (!succeeded)
     {
+        // Free all the file names, if allocated.
         for (int i = 0; i < logFileSize; ++i)
```

Comment:
note: technically speaking, a caller shouldn't do *anything* with out parameters on failure.

```c
int *p
if (!foo(&p)) {
  // shouldn't reference p!!!
}
```

So L219's "done" really shouldn't do anything if the call on L219 failed.